proof-defns/proof_defns.tex

```diff
@@ -0,0 +1,438 @@
+\end{lemma}
+
+\begin{lemma}
+    For any two vectors $v, w \in$ \texttt{SizedDomain(D)}, $d_{Sym} = 2$ if and only if we can change one element of $v$ to obtain $v'$ such that $\MultiSet(v') = \MultiSet(w)$.
+\end{lemma}
+
+The above lemma follows directly from the definition of symmetric distance. 
+
+\begin{lemma}
+Given a function $f$ on input domain \texttt{SizedDomain(D)}, if $\dout(f(v), f(w)) \leq c$ for all vectors $v, w$ such that $d_{Sym}(v, w) = 2$, then $f$ is $c/2$-stable.
+\end{lemma}
+
+By the definition of $c$-stable (Definition~\ref{def:c}), 
+that $f$ is $c/2$-stable is equivalent to stating that for all pairs $v', w'$ in the input domain, the following holds:
+\[
+    d_{\Y}(f(v'), f(w')) \leq c/2 \cdot d_{Sym}(v', w').
+\]
+Also equivalently, this means that the stability relation of $f$ is
+\[
+    \dout \geq c/2 \cdot \din.
+\]
+
+\todo{Add the more general version of path property by using shortest path distance on graphs as discussed on July 19. The proofs also need to be added.}
+
+\subsection{The path property: a generalization}
+
+\subsection{Substitute distance}
+\begin{definition}[Substitute distance]
+We only define the \emph{substitute distance} $d_{Sym}$ on multisets with the same number of elements. On two multisets $u,v\in \MultiSet(\mathcal{X})$ for some domain $\mathcal{X}$ where $|u| = |v|$, we say that the substitute distance $d_{Subs}(u,v)$ is equal to the cardinality of the relative complement $u \backslash v$, so $d_{Subs}(u,v) = |u\backslash v|$. (This can be thought of as fixing multiset $u$ and finding how many elements in $v$ are not represented in $v$.) \silvia{List of domains? Associated types? It is not in the library, so we cannot answer this -- maybe we should drop it.}
+
+\connor{Substitute distance should have the same domain as symmetric distance since it is defined as half the symmetric distance. Probably the same ``associated type'', too, but I don't think we need to cover that here.}
+
+Alternatively, we can define $d_{Subs}$ as
+
+
+$$d_{Subs}(u,v) = \frac{1}{2}d_{\text{Sym}}(u,v).$$
+
+Note that this metric is like a generalization of Hamming distance to multisets (recall that Hamming distance is defined on ordered objects). \silvia{If we say ``recall" then Hamming distance should be defined and explained.} \todo{Add Hamming? It is never used, so I would drop it.}
+\end{definition}
+\silvia{I would not define $d_{Subs}$ in terms of $d_{Sym}$; and instead write original definition and then add the 2-factor equality as a claim.}
+\connor{I think they had defined $d_{subs}$ in terms of $d_{sym}$.}
+
+\begin{claim}
+Substitute distance is a metric.
+\end{claim}
+
+%\begin{claim}
+%(2 factor of Subs and Sym)
+%\end{claim}
+
+\begin{remark}
+    As of June 24, substitute distance has been removed from the library as a possible dataset metric. Therefore, symmetric distance is currently the only available dataset metric in the OpenDP library.
```

Comment:
Substitute distance is still in the library. There is a constructor the converts the metric. It accepts an input metric of SubstituteDistance.

Reply:
Oh okay! Then does it have a list of compatible domains and associated types?